ecl/data_source_ecl_network_qos_options_v2_mock_test.go

```diff
@@ -0,0 +1,154 @@
+package ecl
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/nttcom/terraform-provider-ecl/ecl/testhelper/mock"
+)
+
+func TestMockedAccNetworkV2QosOptionsDataSource_basic(t *testing.T) {
+	mc := mock.NewMockController()
+	defer mc.TerminateMockControllerSafety()
+
+	postKeystone := fmt.Sprintf(fakeKeystonePostTmpl, mc.Endpoint(), OS_REGION_NAME)
+	mc.Register(t, "keystone", "/v3/auth/tokens", postKeystone)
+	mc.Register(t, "qos_options", "/v2.0/qos_optionss", testMockNetworkV2QosOptionsListNameQuery)
+
+	mc.StartServer(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccNetworkV2QosOptionsDataSourceBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNetworkV2QosOptionsDataSourceID("data.ecl_network_qos_options_v2.qos_options_1"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_network_qos_options_v2.qos_options_1", "name", "20M-GA-AZURE"),
+				),
+			},
+		},
+	})
+}
+
+func TestMockedAccNetworkV2QosOptionsDataSource_queries(t *testing.T) {
+	mc := mock.NewMockController()
+	defer mc.TerminateMockControllerSafety()
+
+	postKeystone := fmt.Sprintf(fakeKeystonePostTmpl, mc.Endpoint(), OS_REGION_NAME)
+	mc.Register(t, "keystone", "/v3/auth/tokens", postKeystone)
+	mc.Register(t, "qos_options", "/v2.0/qos_optionss", testMockNetworkV2QosOptionsListNameQuery)
+	mc.Register(t, "qos_options", "/v2.0/qos_optionss", testMockNetworkV2QosOptionsListIDQuery)
+	mc.Register(t, "qos_options", "/v2.0/qos_optionss", testMockNetworkV2QosOptionsListDescriptionQuery)
+
+	mc.StartServer(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccNetworkV2QosOptionsDataSourceName,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNetworkV2QosOptionsDataSourceID("data.ecl_network_qos_options_v2.qos_options_1"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccNetworkV2QosOptionsDataSourceID,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNetworkV2QosOptionsDataSourceID("data.ecl_network_qos_options_v2.qos_options_2"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccNetworkV2QosOptionsDataSourceDescription,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNetworkV2QosOptionsDataSourceID("data.ecl_network_qos_options_v2.qos_options_1"),
+				),
+			},
+		},
+	})
+}
+
+var testMockNetworkV2QosOptionsListNameQuery = `
+request:
+    method: GET
+    query:
+      name:
+        - 20M-GA-AZURE
```

Comment:
実在するQOSオプションを指定して頂きたいので`10Mbps-BestEffort`をご利用頂けたらと思います。
```suggestion
        - 10Mbps-BestEffort
```